source/extensions/omni.isaac.lab_tasks/omni/isaac/lab_tasks/utils/wrappers/torchrl/torchrl_env_wrapper.py

```diff
@@ -0,0 +1,601 @@
+        if next_preset is not None:
+            # tensordict could already have a "next" key
+            # this could be done more efficiently by not excluding but just passing
+            # the necessary keys
+            next_tensordict.update(next_preset.exclude(*next_tensordict.keys(True, True)))
+        tensordict.set("next", next_tensordict)
+        # Information from the "extras" dict of dicts passed from IsaacLab contains reward and metrics information.
+        # The _data postfix is added to distinguish the field containing the actual values from the dummy {key} field
+        # in the tensordict required for faster rollouts. To get logging to work, we need to move the data from the "next" tensordict
+        # to the top level tensordict.
+        # TODO: Find a more elegant solution to enable rewards and metrics logging. "
+        for key in self.env.unwrapped.extras:
+            tensordict[f"{key}_data"] = tensordict["next"][f"{key}_data"]
+        return tensordict
+
+    def _step(self, tensordict: TensorDictBase) -> TensorDictBase:
+        action = tensordict.get(self.action_key)
+        reward = 0
+        for _ in range(self.wrapper_frame_skip):
+            (
+                obs,
+                _reward,
+                terminated,
+                truncated,
+                done,
+                info_dict,
+            ) = self._output_transform(self._env.step(action))
+
+            if _reward is not None:
+                reward = reward + _reward
+
+            terminated, truncated, done, do_break = self.read_done(
+                terminated=terminated, truncated=truncated, done=done
+            )
+            if do_break:
+                break
+        reward = self.read_reward(reward)
+        obs_dict = self.read_obs(obs)
+        obs_dict[self.reward_key] = reward
+
+        # if truncated/terminated is not in the keys, we just don't pass it even if it
+        # is defined.
+        if terminated is None:
+            terminated = done
+        if truncated is not None:
+            obs_dict["truncated"] = truncated
+
+        obs_dict["done"] = done
+        obs_dict["terminated"] = terminated
+        validated = self.validated
+        if not validated:
+            tensordict_out = TensorDict(obs_dict, batch_size=tensordict.batch_size)
+            if validated is None:
+                # check if any value has to be recast to something else. If not, we can safely
+                # build the tensordict without running checks
+                self.validated = all(
+                    val is tensordict_out.get(key) for key, val in TensorDict(obs_dict, []).items(True, True)
+                )
+        else:
+            tensordict_out = TensorDict._new_unsafe(
+                obs_dict,
+                batch_size=tensordict.batch_size,
+            )
+        if self.device is not None:
+            tensordict_out = tensordict_out.to(self.device)
+
+        self._curr_ep_len += 1
```

Comment:
I think that up until here everything is more or less like the `GymWrapper._step`, maybe `super()` would be more apporpriate?